### PR TITLE
add extensions to the interface for Certificate

### PR DESCRIPTION
**Pull request by reaperhulk**
as noted in #1956 this was missing

### CODE REVIEW

**Comment by alex** [comment]
Is it documented?

**Comment by reaperhulk** [comment]
Yeah we documented it here: https://cryptography.io/en/latest/x509/#cryptography.x509.Certificate.extensions

File: src/cryptography/x509.py [hunk 1]
```diff
@@ -1125,6 +1125,12 @@ def signature_hash_algorithm(self):
         in the certificate.
         """
 
+    @abc.abstractproperty
+    def extensions(self):
+        """
+        Returns a list of extensions
```
Comment (alex) [inline src/cryptography/x509.py]: This should be a "sequence" or something, it's not areally a list.

Comment (reaperhulk) [inline src/cryptography/x509.py]: Should probably just say an Extensions object right? :)

Comment (alex) [inline src/cryptography/x509.py]: That also works :-)

On Fri, May 15, 2015 at 12:32 PM, Paul Kehrer <notifications@github.com>
wrote:

> In src/cryptography/x509.py
> <https://github.com/pyca/cryptography/pull/1960#discussion_r30423803>:
>
> > @@ -1125,6 +1125,12 @@ def signature_hash_algorithm(self):
> >          in the certificate.
> >          """
> >
> > +    @abc.abstractproperty
> > +    def extensions(self):
> > +        """
> > +        Returns a list of extensions
>
> Should probably just say an Extensions object right? :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/1960/files#r30423803>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084

**Comment by etrauschke** [comment]
Just mentioning that CSRs have extensions too. But I guess we can skip them for now since we don't have a backend implementation for them anyway.

**Comment by reaperhulk** [comment]
Yeah we'll add them to the interface when we implement support for them (coming soon). We actually have a test that verifies interfaces so if we added it now CI would fail :)

**Pull request merged**